backend/src/v5/models/tickets.groups.js

```diff
@@ -33,8 +44,13 @@ Groups.deleteGroups = async (teamspace, project, model, ticket, groupIds) => {
 	await deleteMany(teamspace, GROUPS_COL, { teamspace, project, model, ticket, _id: { $in: groupIds } });
 };
 
-Groups.getGroupsByIds = (teamspace, project, model, ticket, groupIds, projection) => find(
-	teamspace, GROUPS_COL, { teamspace, project, model, ticket, _id: { $in: groupIds } }, projection);
+Groups.getGroupsByIds = async (teamspace, project, model, ticket, groupIds, projection) => {
+	let groups = await find(teamspace, GROUPS_COL,
+		{ teamspace, project, model, ticket, _id: { $in: groupIds } }, projection);
+
+	groups = await Promise.all(groups.map(convertGroupRules));
```

Comment:
again, no reason to assign groups as a `let` just to assign it then immediately return it.